apollo-gradle-plugin/src/main/java/com/apollographql/android/gradle/ApolloExtension.java

```diff
@@ -1,8 +1,14 @@
 package com.apollographql.android.gradle;
 
+import groovy.lang.Closure;
+
+import java.util.HashMap;
+import java.util.Map;
+
 public class ApolloExtension {
   static final String NAME = "apollo";
   private boolean generateClasses = false;
+  private Map<String, String> customTypeMapping = new HashMap<>();
```

Comment:
Not quite agree with the author about `you’re introducing needless nondeterminism`, if you need the order then for sure `LinkedHashMap` is the right choice but if order doesn't matter for you and primitives used as key for hash then I don't see any issues with `nondeterminism`.

It's not principle for me, will change to `LinkedHashMap`